File: typescript/packages/bench/src/rules/repairDurationRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class RepairDurationRule implements RuleContract {
    readonly key = "repair-duration";

    evaluate(data: Dataset): boolean {
        return data.labourMinutes() > 0;
    }
}

File: rust/crates/atelier-bench/src/rules/repair_duration_rule.rs
//! Rule 10/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct RepairDurationRule;

impl RuleContract for RepairDurationRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        data.labour_minutes() > 0
    }
}
